invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/PatronDetails/components/PatronDocumentRequests/PatronDocumentRequests.js

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Loader, Error } from '../../../../../common/components';
+import { documentRequest as documentRequestApi } from '../../../../../common/api/';
+
+import { ResultsTable } from '../../../../../common/components';
+import { BackOfficeRoutes } from '../../../../../routes/urls';
+import { formatter } from '../../../../../common/components/ResultsTable/formatters';
+import { SeeAllButton } from '../../../components/buttons';
+import { goTo, goToHandler } from '../../../../../history';
+
+import isEmpty from 'lodash/isEmpty';
+import pick from 'lodash/pick';
+
+export default class PatronDocumentRequests extends Component {
+  constructor(props) {
+    super(props);
+    this.fetchPatronDocumentRequests = props.fetchPatronDocumentRequests;
+    this.showDetailsUrl = BackOfficeRoutes.documentRequestDetailsFor;
+    this.seeAllUrl = BackOfficeRoutes.documentRequestsListWithQuery;
+  }
+
+  componentDidMount() {
+    const patronPid = this.props.patronPid ? this.props.patronPid : null;
+    this.fetchPatronDocumentRequests(patronPid);
```

Comment:
how this action will handle the `patronPid = null`?

Reply:
Not well after looking into it... I thought it was supposed to be handled when building the query with `withPatronPid` but since I used the query building inside the `try { .. } catch` it caught that error too. I will move the query builder outside and it's behaving properly now. Nice catch!